shared/common-utils.ts

```diff
@@ -99,3 +101,11 @@ function splitIntoChunks(s: string, chunkSize: number): string[] {
 export function addDigitSeparator(n: string, digitSeparator: string, chunkSize: number): string {
     return splitIntoChunks(n, chunkSize).join(digitSeparator);
 }
+
+export function splitArguments(options = ''): string[] {
+    // escape hashes first, otherwise they're interpreted as comments
```

Comment:
Maybe this is an indication this isn't the best package to use? It also seems it supports control characters https://github.com/ljharb/shell-quote/blob/main/parse.js

Can you add tests for how stuff like operators and piping behaves? https://github.com/ljharb/shell-quote/blob/main/test/op.js#L6

Reply:
I'll add them in a separate PR as this was almost purely code motion and I don't want to further muddy it.

Reply:
> Maybe this is an indication this isn't the best package to use? It also seems it supports control characters https://github.com/ljharb/shell-quote/blob/main/parse.js

I don't know that we want to support those? Maybe? It would be a change in behaviour for anyone with `\` in them (maybe Windows people?)

Reply:
Oh! That's 100% your point!!! :'( yes...

Reply:
Changing anything about this changes the API, people have never complained about it. They did complain one time where it got changed to split on space. Unless someone has a valid use-case for different behaviour, please do not change it.

Reply:
Don't worry, I added tons of tests and kept all the behaviour the same _except where it was absolutely bonkers_ before. (like the weird handling of pipe characters and the escaping of quote strings ?!).

The PR will be up for a while anyway. We can't be too afraid of breaking things that are beyond undocumented like "what happens if you pass an argument that has a pipe character followed by double backslash and a hash")